packages/kbn-config/src/deprecation/apply_deprecations.ts

```diff
@@ -22,9 +23,21 @@ export const applyDeprecations = (
   deprecations: ConfigDeprecationWithContext[],
   createAddDeprecation: (pluginId: string) => AddConfigDeprecation = noopAddDeprecationFactory
 ) => {
-  let processed = cloneDeep(config);
+  const result = cloneDeep(config);
   deprecations.forEach(({ deprecation, path }) => {
-    processed = deprecation(processed, path, createAddDeprecation(path));
+    const commands = deprecation(result, path, createAddDeprecation(path));
+    if (commands) {
+      if (commands.set) {
```

Comment:
why not use `commands?.set` instead of the two if statements?

Reply:
ok, I will change in a follow-up